src/foam/core/reflow/AbstractDAOAgent.js

```diff
@@ -278,23 +278,24 @@ foam.CLASS({
       this.columns$ = this.block.value.columns$;
       this.block.value.value = this;
 
-      e.add(this.dynamic(function(columns) {
-        var config = {
-          data: self.unlimitedDAO,
-          config: self.DAOControllerConfig.create({
-            dao: self.unlimitedDAO,
-            disableSelection: false
-          })
-        };
+      var config = {
+        data: self.unlimitedDAO,
+        config: self.DAOControllerConfig.create({
+          dao: self.unlimitedDAO,
+          disableSelection: false
+        })
+      };
 
-        if ( columns.length ) {
-          config.selectedColumnNames = columns;
-        }
+      if ( self.columns.length && self.columns != '*' ) {
+        config.selectedColumnNames = self.columns;
+      } else {
+        // reset columns
+        localStorage.removeItem(self.dao.of.id);
```

Comment:
I was thinking about this. Perhaps we export 'localStorage' from Window.js, but we actually export it twice, once as localStorage and once as columnStorage and then in the table view we use columnStorage instead.
Then the DAOPrompt can export it's own columnStorage so that it doesn't interfere with the system default. Also, when setting or getting the columns, we should use the explict localStorage api rather than just treating it as a map, that way it's easier to implement an alternative implementation: https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage

Reply:
thanks, will try